### PR TITLE
Feature: Add translations via plugin application bootstrap

### PR DESCRIPTION
**Pull request by Draykee**
Pull Request for #810 

**Change List:**
- Extended `I18nService` with `addTranslationFile` and `addTranslation`
- Created `translations.e2e-spec.ts` with TranslationTestPlugin
- Changed deprecated[ i18next-node-fs-backend](https://github.com/i18next/i18next-node-fs-backend) to [i18next-fs-backend](https://github.com/i18next/i18next-fs-backend)
- Exports `I18nService` and `I18nError` in public api
- Fixed language detection of i18next module (changed to `lookupQuerystring: 'languageCode'`)
- Added documentation to developer guide and I18nService typescript documentation (used for linking)

### CODE REVIEW

**Comment by Draykee** [comment]
@michaelbromley I think this PR should be done and reviewable for you (or anyone else) 🥳

File: packages/core/src/i18n/i18n.service.ts [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
 import path from 'path';
 
 import { GraphQLErrorResult } from '../common/error/error-result';
+import { Logger } from '../config';
 import { ConfigService } from '../config/config.service';
 
 import { I18nError } from './i18n-error';
 
+/**
+ * I18n resources used for translations
```
Comment (michaelbromley) [inline packages/core/src/i18n/i18n.service.ts:19]: Needs a `@description` tag to be picked up by docs generator.

File: packages/core/src/i18n/i18n.service.ts [hunk 2]
```diff
@@ -21,11 +33,19 @@ export interface I18nRequest extends Request {
  * The `i18next-express-middleware` middleware detects the client's preferred language based on
```
Comment (michaelbromley) [inline packages/core/src/i18n/i18n.service.ts:35]: Same on the `@description` tag

File: packages/core/src/i18n/i18n.service.ts [hunk 3]
```diff
@@ -44,12 +64,40 @@ export class I18nService implements OnModuleInit {
...
+            const rawData = fs.readFileSync(filePath);
+            const resources = JSON.parse(rawData.toString('utf-8'));
+            this.addTranslation(langKey, resources);
+        } catch (err) {
+            Logger.error(`Could not load resources file ${filePath}`, `I18nService`);
+        }
+    }
+
+    /**
+     * Add a I18n translation file
```
Comment (michaelbromley) [inline packages/core/src/i18n/i18n.service.ts]: not a "file" in this case right?

**Review by michaelbromley — COMMENTED** [review]
Absolutely excellent PR - docs, e2e tests, implementation, 😗 👌 

Just missing some `@description` tags on the JSDoc blocks so that they show up in the docs.

File: packages/core/src/i18n/i18n.service.ts (hunk 3 above)
Comment (kevmtt) [reply packages/core/src/i18n/i18n.service.ts]: Yep. Good old copy+paste error ^^

**Comment by michaelbromley** [comment]
Thank you!

**Pull request merged**